source/coap_message_handler.c

```diff
@@ -74,6 +74,7 @@ static coap_transaction_t *transaction_create(void)
     if (this) {
         memset(this, 0, sizeof(coap_transaction_t));
         this->client_request = true;// default to client initiated method
+        this->have_local_address = false;
```

Comment:
Still think flag is code overhead you don't. Source address "any" should just work. (If it doesn't, bug me to fix it).